src/App.tsx

```diff
@@ -1,54 +1,92 @@
-import React from 'react';
+import React, { useCallback, useRef, useState } from 'react';
+import debounce from 'lodash.debounce';
+import classNames from 'classnames';
+
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const prevQuery = useRef('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [openMenu, setOpenMenu] = useState(false);
+
+  const apllyQuery = useCallback(
+    debounce(setAppliedQuery, 1000),
+    [],
+  );
+
+  const handleQueryCahnge = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
```

Comment:
you did setQuery after apllyQuery, and after check if input value the same as previous search, looks strange, better to check it before